File: gdx2d-desktop/src/hevs/gdx2d/demos/music/midiplayer/MIDIListener.java
package hevs.gdx2d.demos.music.midiplayer;

import java.util.ArrayList;

/**
 * Interface for communication between classes
 *
 * @author Mikael Follonier (fom)
 * @version 1.0
 */
public interface MIDIListener {
	/**
	 * Method called when a MIDI message has to be handled
	 * @param msg
	 */
	void onMIDIMessage(ArrayList<MIDIMessage> msg);

	/**
	 * Method called when the MIDI track is finished
	 */
	void onTrackFinished();
}

File: gdx2d-demoDesktop/src/main/scala/ch/hevs/gdx2d/demos/music/midiplayer/midilayer/MIDIListener.scala
package ch.hevs.gdx2d.demos.music.midiplayer.midilayer

import java.util.ArrayList

/**
 * Interface for communication between classes.
 *
 * @author Mikael Follonier (fom)
 * @version 2.0
 */
trait MIDIListener {
  /**
   * Method called when a MIDI message has to be handled.
   */
  def onMIDIMessage(msg: ArrayList[MIDIMessage]): Unit

  /**
   * Method called when the MIDI track is finished.
   */
  def onTrackFinished(): Unit
}
